### PR TITLE
feat(isOdd): simplify code implementation

### PR DESCRIPTION
**Pull request by kelvinice**
- Added extensive validation for input types and special cases (e.g., NaN, Infinity, zero).
- Introduced binary conversion to determine oddness
- Implemented a chaos factor with a 1% chance to always return true, embracing randomness.
- Added verbose logging for each step, including philosophical commentary about zero.
- Ensured maximum absurdity while preserving functionality.

### CODE REVIEW

**Review by brandon-julio-t — APPROVED** [review]
lgtm 👍

File: index.ts [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+      throw new Error('Invalid input: Numbers only, please. No strings, no arrays, no cats.');
+  }
+
+  // Step 2: Handle special cases. Because edge cases love to ruin your day.
+  if (isNaN(number)) {
+      throw new Error('Invalid input: NaN is neither odd nor even. It’s just... NaN.');
+  }
+  if (!Number.isFinite(number)) {
+      throw new Error('Invalid input: Infinity cannot be odd or even. It’s just infinite.');
+  }
```
Comment (cipher416) [inline index.ts:13]: how does this affect lebron's legacy?

**Review by cipher416 — CHANGES_REQUESTED** [review]

**Comment by cipher416** [comment]
image

File: index.ts [hunk 2]
```diff
@@ -1,3 +1,44 @@
-export default function isOdd(number: number) : boolean {
-  return number % 2 != 0;
+export default function isOdd(number: number): boolean {
+  // Step 1: Validate input. Because sanity matters in an insane world.
+  if (typeof number !== 'number') {
+      throw new Error('Invalid input: Numbers only, please. No strings, no arrays, no cats.');
```
Comment (cipher416) [inline index.ts:4]: please i18n

**Comment by kelvinice** [comment]
@felixgoz merge it asap, or else I will steal your balls

**Comment by felixgoz** [comment]
I see an element of gambling in Step 7. I enjoy gambling, so I accept it.

File: index.ts [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+  console.log(
+      `Least significant bit analysis says: ${isOdd ? 'Odd!' : 'Even!'}`
+  );
+
+  // Step 7: Add a random chance of chaos. Who doesn’t love randomness?
+  const chaosFactor = Math.random() > 0.99;
+  if (chaosFactor) {
+      console.log('The Chaos Gods have spoken. Everything is odd now.');
+      return true;
+  }
```
Comment (cipher416) [inline index.ts:40]: please write a test for this step

Comment (felixgoz) [reply index.ts:40]: lgtm 👍

**Pull request merged**